nodejs.py

```diff
@@ -0,0 +1,234 @@
+        self._total_size = size
+        io.FileIO.__init__(self, path, *args, **kwargs)
+
+    def write(self, b):
+        progress = min(100.0, ReadProgressFileObject.current_read / (self._total_size * 0.01))
+        sys.stdout.write("\r[%3.2f%%] " %(progress))
+        sys.stdout.flush()
+        return io.FileIO.write(self, b)
+
+Command = collections.namedtuple("Command", "aliases function")
+DepsDirectory = collections.namedtuple("DepsDirectory", "path include")
+
+# Command-Line setup
+parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter)
+
+#-----------------------------------------------------------------------
+def flush_cache(silent = False):
+    if not silent:
+        print "[flush-cache]"
+
+    utils.store_nodejs_output(None, ".")
+
+    if not silent:
+        print "Done" 
+
+cmd_flush_cache = Command(
+    aliases=["flush-cache", "fc"],
+    function=flush_cache,
+)
+#-----------------------------------------------------------------------
+def git_init():
+    print "[git-init]"
+
+    # TODO: add CLI overide options
+    # - Node version
+    # - J2V8 version
+
+    utils.store_nodejs_output(None, ".")
+
+    if (not os.path.exists("node")):
+        print "Cloning Node.js version: " + settings.NODE_VERSION
+        # NOTE: autocrlf=false is very important for linux based cross-compiles of Node.js to work on a windows docker host
+        utils.execute("git clone https://github.com/nodejs/node --config core.autocrlf=false --depth 1 --branch v" + settings.NODE_VERSION)
+    else:
+        print "Node.js is already cloned & checked out"
+        apply_diff(True)
+
+    print "Done"
+
+cmd_git_init = Command(
+    aliases=["git-init", "gi"],
+    function=git_init
+)
+#-----------------------------------------------------------------------
+def package():
+    print "[package]"
+
+    platforms = sys.argv[2:]
+    full = len(platforms) == 0
+
+    # make sure all node.js binaries are stored in the cache before packaging
+    flush_cache(True)
+
+    # C++ header files
+    # NOTE: see https://stackoverflow.com/a/4851555/425532 why this weird syntax is necessary here
+    dependencies = {
+        "list": [
+            DepsDirectory(path="./node/deps/", include=[".h"]),
+            DepsDirectory(path="./node/src/", include=[".h"]),
+        ],
+        "size": 0,
+    }
+
+    def __add_platform_deps(platform, include, vendor = None):
+        target = bc.platform_configs.get(platform)
+        vendor_str = (vendor + "-" if vendor else "")
+        selected = (vendor_str + platform) in platforms
+
+        if (full or selected):
+            dependencies["list"] += [
+                DepsDirectory(
+                    path="./node.out/" + vendor_str + platform + "." + arch + "/",
+                    include=["j2v8.node.out"] + include
+                ) for arch in target.architectures
+            ]
+
+    # speciffy the platforms & file patterns that should be included
```

Comment:
Yeah, I put any further documentation in BUILDING.md aside intentionally, because the overall process is not yet fully drafted out yet ... but it should now not be far off that I can write a more extensive documentation that describes all the bells and whistles 😄